src/source/CppGenerator.scala

```diff
@@ -158,12 +158,20 @@ class CppGenerator(spec: Spec) extends Generator(spec) {
       })
   }
 
-  def generateHppConstants(w: IndentWriter, consts: Seq[Const], exportHeaderAdded: Boolean = false) = {
+  // To export symbols in windows dll
+  def isExportHeaderNeeded() : Boolean = {
+    return spec.exportHeaderName.length > 0;
+  }
+
+  def getExportMacro(): String = {
+    return if (isExportHeaderNeeded()) s"${spec.exportHeaderName.toUpperCase()} " else "";
+  }
+
+  def generateHppConstants(w: IndentWriter, consts: Seq[Const]) = {
     for (c <- consts) {
       w.wl
       writeDoc(w, c.doc)
-      val exportHeader = if (exportHeaderAdded) s"${spec.exportHeaderName.toUpperCase()} " else "";
-      w.wl(s"static ${exportHeader}${marshal.fieldType(c.ty)} const ${idCpp.const(c.ident)};")
+      w.wl(s"static ${marshal.fieldType(c.ty)} const ${idCpp.const(c.ident)};")
```

Comment:
So you removed the `${exportHeader}`?

Reply:
Yes, because now I have it on class definition.